File: app/src/main/java/com/wjf/self_demo/view/EditTextDialog.java
package com.wjf.self_demo.view;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;

import com.wjf.self_demo.R;
import com.wjf.self_demo.databinding.DialogEditBinding;
import com.wjf.self_library.common.BaseDialog;

/**
 * @author WJF
 */
public class EditTextDialog extends BaseDialog<DialogEditBinding> {


    public EditTextDialog(@NonNull Context context) {
        super(context);
    }

    @Override
    protected void initView(DialogEditBinding view) {

    }

    public EditTextDialog setSubmitListener(BaseDialog.DialogClickListener<String> listener){
        view.submitBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onClick(view.inputEdit.getText().toString().trim());
                view.inputEdit.setText("");
                EditTextDialog.this.dismiss();
            }
        });
        return this;
    }

    @Override
    protected int setLayout() {
        return R.layout.dialog_edit;
    }

}

File: app/src/main/java/com/wjf/self_demo/view/EditTextDialog.kt
package com.wjf.self_demo.view

import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.WindowManager
import com.wjf.self_demo.databinding.DialogEditBinding
import org.jxxy.debug.corekit.common.BaseDialog
import org.jxxy.debug.corekit.util.singleClick

/** @author WJF
 */
class EditTextDialog : BaseDialog<DialogEditBinding>() {
    init {
        width = WindowManager.LayoutParams.MATCH_PARENT
        height = WindowManager.LayoutParams.WRAP_CONTENT
        ifCancelOnTouch = true
        gravity = Gravity.BOTTOM
    }

    var listener: ((String) -> Unit)? = null
    fun setSubmitListener(block: (String) -> Unit): EditTextDialog {
        this.listener = block
        return this
    }

    override fun initView() {
        view.submitBtn.singleClick { v: View? ->
            listener?.invoke(view.inputEdit.text.toString().trim { it <= ' ' })
            view.inputEdit.setText("")
            dismiss()
        }
    }

    override fun bindLayout(inflater: LayoutInflater): DialogEditBinding {
        return DialogEditBinding.inflate(inflater)
    }
}
